torch/csrc/distributed/rpc/tensorpipe_utils.cpp

```diff
@@ -177,9 +181,12 @@ std::tuple<tensorpipe::Message, TensorpipeWriteBuffers> tensorpipeSerialize(
   tpMessage.payloads.push_back(tensorpipe::Message::Payload{
       buffers.pickle.data(), buffers.pickle.size()});
   const std::vector<torch::Tensor>& tensorDataVec = pickler.tensorData();
-  tpMessage.tensors.reserve(tensorDataVec.size());
-  for (const auto i : c10::irange(tensorDataVec.size())) {
-    const torch::Tensor& tensor = tensorDataVec[i];
+  std::vector<torch::Tensor> nonMetaTensorDataVec;
```

Comment:
This seems very wasteful. Why not just manually take care of the `i` increments in the loop below so you don't have to allocate a temporary vector (with all refcount bumps). I would care less if this only affected meta tensor code but this pessimizes all cases.

Reply:
addressed